users/views.py

```diff
@@ -1,11 +1,44 @@
 import json
+import requests
+import jwt
 
+from django.conf        import settings
+from django.forms       import ValidationError
 from django.http        import JsonResponse
 from django.views       import View
-from django.forms       import ValidationError
+from core.utils         import check_first_name, check_last_name, check_phone_number, signin_decorator
 
 from users.models       import User
-from core.utils         import check_first_name, check_last_name, check_phone_number, signin_decorator
+
+class KakaoOauthView(View):
+    def get(self, request):
+        token = request.META.get('HTTP_AUTHORIZATION')
+
+        KAKAO_URL = 'https://kapi.kakao.com/v2/user/me'
+        HEADER    = {'Authorization': token}
+        
+        user_info = requests.get(KAKAO_URL,headers=HEADER).json()
+        
+        kakao_id          = user_info['id']
+        kakao_profile_img = user_info['kakao_account']['profile']['profile_image_url']
+        email             = user_info['kakao_account']['email']
+    
+        user, created = User.objects.get_or_create(
+            kakao_id = kakao_id,
+            defaults = {
+                'kakao_profile_img': kakao_profile_img,
+                'email'            : email
+            }
+        ) 
+    
+        if not created:
+            user.kakao_proifile_img = kakao_profile_img
+            user.save()
+            
+            token = jwt.encode({'id':user.id}, settings.SECRET_KEY, settings.ALGORITHM)
+            return JsonResponse({'message':'SIGNIN_SUCCESS', 'token':token}, status=200)
+    
+        return JsonResponse({'message':'SIGNUP_SUCCESS'}, status=201)
```

Comment:
기존에 없던 유저가 카카오로 로그인을 시도하더라도 토큰을 줘야겟죠?
```suggestion
        if not created:
            user.kakao_proifile_img = kakao_profile_img
            user.save()
            
        token = jwt.encode({'id':user.id}, settings.SECRET_KEY, settings.ALGORITHM)
    
        return JsonResponse({'message':'SUCCESS', 'token' : token}, status=201 if created else 200)
```

Reply:
저희 DB에 없는 유저가 카카오로 첫 로그인을 시도한다면 회원가입이 됩니다. 
회원가입이 되면, 사용자 정보를 추가로 입력받아야하고 
해당 기능은 로그인 데코레이터를 사용해 토큰으로 유저의 신원을 확인하고 있습니다.

그렇기 때문에 회원가입시에도 토큰 발급이 필요한거구나 라고 생각하고 코드를 수정했습니다!

저는 로그인할때만 토큰을 발급하는 걸로 알고있었는데, 
회원가입시에도 토큰을 주는 이유가 저게 맞는지 궁금합니다!